cmd/ccboc/main.go

```diff
@@ -121,6 +123,117 @@ func getCalculationID(c *cli.Context) error {
 	return nil
 }
 
+func getCalculationResult(c *cli.Context) error {
+	teff := c.Float64("teff")
+	logG := c.Float64("logG")
+	path := c.String("results-download-path")
+
+	resp, err := http.Get(globalConfig.APIURL + "/calculations/results?teff=" + fmt.Sprintf("%0.1f", teff) + "&logg=" + fmt.Sprintf("%0.2f", logG))
+	if err != nil {
+		logrus.WithError(err).Fatal("Couldn't retrieve the data from headers.")
+	}
+
+	key := resp.Header.Get("Content-Disposition")
+	newKey := regexp.MustCompile("=").Split(key, 2)
+
+	var defaultPath string
+
+	user, err := user.Current()
+	if err != nil {
+		logrus.WithError(err).Fatal("couldn't get $HOME directory from current user")
+	}
+	home := user.HomeDir
+
+	if path != "" {
+		defaultPath = filepath.Join(path, filepath.Base(newKey[1]))
+	} else {
+		logrus.Info("Desired path to download the calculations was not included. The calculations are set to be downloaded into: $HOME/.config/ccboc/{id}")
+		defaultPath = filepath.Join(home, "/.config/ccboc/", filepath.Base(newKey[1]))
+	}
+
+	if _, err := os.Stat(defaultPath); !errors.Is(err, os.ErrNotExist) {
+		logrus.Info("Desired path to download the calculations does not exist. The calculations are set to be downloaded into: $HOME/.config/ccboc/{id}")
+		defaultPath = filepath.Join(home, "/.config/ccboc/", filepath.Base(newKey[1]))
+	}
+
+	body, responseError, err := request("GET", globalConfig.APIURL+"/calculations/results?teff="+fmt.Sprintf("%0.1f", teff)+"&logg="+fmt.Sprintf("%0.2f", logG), bytes.NewBuffer(nil))
+	if err != nil {
+		logrus.WithError(err).Fatal("error while perfoming the http request")
+	} else if responseError != nil {
+		logrus.WithFields(logrus.Fields{"message": responseError.Message, "status_code": responseError.StatusCode}).Fatal("errors occurred")
+	}
+
+	file, err := os.Create(defaultPath)
+	if err != nil {
+		logrus.WithError(err).Fatal("Couldn't create the calculation result file.")
+	}
+
+	defer file.Close()
+
+	_, err = file.Write(body)
+	if err != nil {
+		logrus.WithError(err).Fatal("Couldn't write the data into the tar file.")
+	}
+
+	return nil
+}
+
+func getCalculationResultByID(c *cli.Context) error {
+	args := os.Args
+	calcID := args[len(args)-1]
+
+	path := c.String("results-download-path")
+
+	resp, err := http.Get(globalConfig.APIURL + "/calculations/results/" + calcID)
+	if err != nil {
+		logrus.WithError(err).Fatal("Couldn't retrieve the data from headers.")
+	}
+
+	key := resp.Header.Get("Content-Disposition")
+	newKey := regexp.MustCompile("=").Split(key, 2)
+
+	var defaultPath string
+
+	user, err := user.Current()
+	if err != nil {
+		logrus.WithError(err).Fatal("couldn't get $HOME directory from current user")
+	}
+	home := user.HomeDir
+
+	if path != "" {
+		defaultPath = filepath.Join(path, filepath.Base(newKey[1]))
+	} else {
+		logrus.Info("Desired path to download the calculations was not included. The calculations are set to be downloaded into: $HOME/.config/ccboc/{id}")
```

Comment:
no need for this log. Instead, just log in where we downloaded the results.

Reply:
i think the user should know, that the path was not added properly and was not found.